DNNCommunity/Dnn.CommunityForums
# Resolve numerous ForumViewer issues

### Description of PR...
Operations from ForumViewer module need to stay on that module rather than redirect to forums module (i.e. on different page)

## Changes made
- Internal changes to determine forums ModuleId when running from forums viewer module
- Incorporates #396 

## How did you test these updates?  

1. Create a page with the Forums module on it
2. Create another page with the ForumsViewer module on it
3. Link the viewer to a forum from the main module
4. Create posts and replies in the viewer module
5. Ensure that all operations keep you within the viewer module
6. Also tested 'my settings', 'active posts', etc.

## PR Template Checklist



## Please mark which issue is solved


Close #373
